### PR TITLE
feat: add NanoLimbo

### PR DESCRIPTION
**Pull request by NgLoader**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### New Server Submissions:

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your server use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the server to the main README.md?
4. [x] Have you added a unique README.md for the server you are adding?

Requested: https://github.com/parkervcp/eggs/issues/1409

### CODE REVIEW

File: game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        "version": "PTDL_v1",
+        "update_url": null
+    },
+    "exported_at": "2021-11-24T20:42:37+01:00",
+    "name": "NanoLimbo",
+    "author": "mail@wuffy.eu",
+    "description": "This is lightweight minecraft limbo server, written on Java with Netty. The main goal of the project is maximum simplicity with a minimum number of sent and processed packets. This limbo is empty, there are no ability to set schematic building since this is not necessary. You can send useful information in chat or BossBar.\r\n\r\nNo plugins, no logs. The server is fully clear. It only able keep a lot of players while the main server is down.",
+    "features": null,
+    "images": [
+        "quay.io\/pterodactyl\/core:java"
```
Comment (Software-Noob) [inline game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:13]: This is a very outdated image that's no longer used. Please change to the [yolks repository](https://github.com/pterodactyl/yolks)

File: game_eggs/minecraft/java/nanolimbo/README.MD [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Paper
```
Comment (Software-Noob) [inline game_eggs/minecraft/java/nanolimbo/README.MD:1]: This is not paper

File: README.md [hunk 3]
```diff
@@ -137,6 +137,7 @@ If you are reading this it looks like you are looking to add an egg to your serv
   * [Technic](game_eggs/minecraft/java/technic)
   * [Tuinity](game_eggs/minecraft/java/tuinity)
   * [VanillaCord](game_eggs/minecraft/java/vanillacord)
+  * [NanoLimbo](/game_eggs/minecraft/java/nanolimbo)
```
Comment (Software-Noob) [inline README.md:140]: Please keep all the READMEs in alphabetical order

File: game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        "startup": "{\r\n    \"done\": \"Server started on\"\r\n}",
+        "logs": "{}",
+        "stop": "^C"
+    },
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/bash\r\n# NanoLimbo Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\n\r\napt update\r\napt install -y curl git openssl bash bc apt-transport-https gnupg software-properties-common\r\n\r\nif [ -z \"${DL_PATH}\" ]; then\r\n    DL_PATH=`curl -Ls -o \/dev\/null -w %{url_effective} https:\/\/github.com\/Nan1t\/NanoLimbo\/releases\/latest`\r\n    DL_PATH=$(grep -oP '[0-9].[0-9].[0-9]' <<< \"$DL_PATH\")\r\n    DL_PATH=https:\/\/github.com\/Nan1t\/NanoLimbo\/releases\/download\/v$DL_PATH\/NanoLimbo-$DL_PATH-all.jar\r\nfi\r\n\r\ncd \/mnt\/server\r\n\r\nif [ ! -f \"settings.yml\" ]; then\r\n    echo \"Downloading config file...\"\r\n    curl -sSL -o settings.yml https:\/\/gist.githubusercontent.com\/NgLoader\/c1ec994cc426a2b0ce0b4245ebf9e412\/raw\/38e94ec5c31f987c25688b4bfd007967d983f187\/settings.yml\r\nfi\r\n\r\necho \"Downloading server file...\"\r\ncurl -sSL -o ${SERVER_JARFILE} ${DL_PATH}",
```
Comment (Software-Noob) [inline game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:25]: We actually have an existing [Github release grabber script](https://github.com/parkervcp/eggs/blob/master/scripts/github_release_grabber.sh) that takes advantage of their API

File: game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            "name": "Server jar File",
+            "description": "The name of the server jarfile to run the server with.",
+            "env_variable": "SERVER_JARFILE",
+            "default_value": "server.jar",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|regex:\/^([\\w\\d._-]+)(\\.jar)$\/"
+        },
+        {
+            "name": "Download Path",
```
Comment (Software-Noob) [inline game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:41]: This name is misleading. I would instead name it Download URL and specify in the description that it's optional, although unnecessary if we use Github releases.

**Review by Software-Noob — CHANGES_REQUESTED** [review]

File: game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json (hunk 1 above)
Comment (NgLoader) [reply game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:13]: Okay

File: game_eggs/minecraft/java/nanolimbo/README.MD (hunk 2 above)
Comment (NgLoader) [reply game_eggs/minecraft/java/nanolimbo/README.MD:1]: Oh, sry

File: game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json (hunk 4 above)
Comment (NgLoader) [reply game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:25]: nice

Comment (NgLoader) [reply game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:25]: @Software-Noob How do I get the download link?

```java
export GITHUB_PACKAGE="Nan1t/NanoLimbo"
export MATCH="v"
curl -s https://raw.githubusercontent.com/parkervcp/eggs/master/scripts/github_release_grabber.sh | bash
```

Comment (Software-Noob) [reply game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:25]: Copy the contents of the script into the install script of the egg itself. There's no reason to pass it to bash.

Comment (NgLoader) [reply game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:25]: Okay

**Comment by Software-Noob** [comment]
We'd probably want to add the `VERSION` variable that already exists in the Github release grabber allowing users to specify a version if they don't want the latest.  This would also require deleting the line 10 `echo "Using latest version"`

For troubleshooting when users report issues, please print the URL it's going to use `echo -e "performing curl -sSL ${DOWNLOAD_URL}"`

A validator for the provided download URL would be good to have before proceeding with downloading anything, or it might output HTML errors into jar file when the link is invalid instead of downloading the real actual jar file:

```
if [ ! -z "${DOWNLOAD_URL}" ]; then
    if curl --output /dev/null --silent --head --fail ${DOWNLOAD_URL}.jar; then
        echo -e "Provided download link is valid. Proceeding to download"
    else
        echo -e "Provided download link is invalid. Exiting now"
        exit 2
    fi
fi
```
Other than that, LGTM.

File: game_eggs/minecraft/java/nanolimbo/README.MD [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# NanoLimbo
+
+This is lightweight minecraft limbo server, written on Java with Netty. The main goal of the project is maximum simplicity with a minimum number of sent and processed packets. This limbo is empty, there are no ability to set schematic building since this is not necessary. You can send useful information in chat or BossBar.
```
Comment (ajgeiss0702) [inline game_eggs/minecraft/java/nanolimbo/README.MD:3]: Please correct this grammar

("This is a lightweight" or preferably "A lightweight")

File: game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
+    "meta": {
+        "version": "PTDL_v1",
+        "update_url": null
+    },
+    "exported_at": "2021-11-24T22:41:34+01:00",
+    "name": "NanoLimbo",
+    "author": "mail@wuffy.eu",
+    "description": "This is lightweight minecraft limbo server, written on Java with Netty. The main goal of the project is maximum simplicity with a minimum number of sent and processed packets. This limbo is empty, there are no ability to set schematic building since this is not necessary. You can send useful information in chat or BossBar.\r\n\r\nNo plugins, no logs. The server is fully clear. It only able keep a lot of players while the main server is down.",
```
Comment (ajgeiss0702) [inline game_eggs/minecraft/java/nanolimbo/egg-nano-limbo.json:10]: Please correct this grammar

("This is a lightweight" or preferably "A lightweight")

File: game_eggs/minecraft/java/README.md [hunk 8]
```diff
@@ -57,6 +57,10 @@ The second method requires you to know the id for both the modpack and version i
 may not get updated with the correct ip address and port at first launch.
 Please restart the server after first launch to fix this.**  
 
+## NanoLimbo
+[NanoLimbo](https://github.com/Nan1t/NanoLimbo)
+This is lightweight minecraft limbo server, written on Java with Netty. The main goal of the project is maximum simplicity with a minimum number of sent and processed packets.
```
Comment (ajgeiss0702) [inline game_eggs/minecraft/java/README.md:62]: Please correct this grammar

("This is a lightweight" or preferably "A lightweight")

**Review by ajgeiss0702 — CHANGES_REQUESTED** [review]

**Comment by NgLoader** [comment]
> We'd probably want to add the `VERSION` variable that already exists in the Github release grabber allowing users to specify a version if they don't want the latest. This would also require deleting the line 10 `echo "Using latest version"`
> 
> For troubleshooting when users report issues, please print the URL it's going to use `echo -e "performing curl -sSL ${DOWNLOAD_URL}"`
> 
> A validator for the provided download URL would be good to have before proceeding with downloading anything, or it might output HTML errors into jar file when the link is invalid instead of downloading the real actual jar file:
> 
> ```
> if [ ! -z "${DOWNLOAD_URL}" ]; then
>     if curl --output /dev/null --silent --head --fail ${DOWNLOAD_URL}.jar; then
>         echo -e "Provided download link is valid. Proceeding to download"
>     else
>         echo -e "Provided download link is invalid. Exiting now"
>         exit 2
>     fi
> fi
> ```
> 
> Other than that, LGTM.

Okay, done

**Review by parkervcp — APPROVED** [review]
LGTM

**Review by Software-Noob — APPROVED** [review]
LGTM, thank you for contributing :+1:

**Pull request merged**